### ISSUE TITLE
Bug report 

### ISSUE THREAD

**Issue by yinghuazh** [azkaban/azkaban#2191]
It was scheduled once, but now it's scheduled every minute.

![image](https://user-images.githubusercontent.com/16507539/56189174-e57c5f80-6059-11e9-9e15-e073c1e97295.png)

**Comment by burgerkingeater** [comment]
Yep, this is a know bug. Thanks for reporting it. Are you interested in
contributing a fix?

**Comment by yinghuazh** [comment]
I don't know what caused it. I don't have the ability to repair it yet. I don't know if it's AK or quartz.

**Comment by yinghuazh** [comment]
@burgerkingeater 

I found that cron had a problem getting NextCheckTime, and if not, the last time it returned
  private long calculateNextCheckTime() {
    DateTime date = new DateTime(this.nextCheckTime).withZone(this.timezone);
    int count = 0;
    while (!date.isAfterNow()) {
      if (count > 100000) {
        throw new IllegalStateException(
            "100000 increments of period did not get to present time.");
      }
      if (this.period == null && this.cronExpression == null) {
        break;
      } else if (this.cronExecutionTime != null) {
        final Date nextDate = this.cronExecutionTime.getNextValidTimeAfter(date.toDate());
        // Some Cron Expressions possibly do not have follow-up occurrences
        if (nextDate != null) {
          date = new DateTime(nextDate);
        } else {
          break;
        }
      } else {
        date = date.plus(this.period);
      }
      count += 1;
    }
    return date.getMillis();
  }

if nextDate is null,returns the value of this. nextCheckTime, which is the last time, so it should have ended the timing task instead of ending it.
Can you change it to the following, if nextDate is null, return Constants. DEFAULT_SCHEDULE_END_EPOCH_TIME

  private long calculateNextCheckTime() {
    DateTime date = new DateTime(this.nextCheckTime).withZone(this.timezone);
    int count = 0;
    while (!date.isAfterNow()) {
      if (count > 100000) {
        throw new IllegalStateException(
            "100000 increments of period did not get to present time.");
      }
      if (this.period == null && this.cronExpression == null) {
        break;
      } else if (this.cronExecutionTime != null) {
        final Date nextDate = this.cronExecutionTime.getNextValidTimeAfter(date.toDate());
        // Some Cron Expressions possibly do not have follow-up occurrences
        if (nextDate != null) {
          date = new DateTime(nextDate);
        } else {
          return Constants.DEFAULT_SCHEDULE_END_EPOCH_TIME;
        }
      } else {
        date = date.plus(this.period);
      }
      count += 1;
    }
    return date.getMillis();
  }

I've tested it. It's OK.

**Comment by burgerkingeater** [comment]
thanks, do you want to submit a PR?

**Comment by yinghuazh** [comment]
I just tried to submit PR, but I couldn't.
